pkgs/swift2objc/lib/src/transformer/transformers/transform_property.dart

```diff
@@ -0,0 +1,108 @@
+import '../../ast/_core/shared/referred_type.dart';
+import '../../ast/declarations/compounds/class_declaration.dart';
+import '../_core/unique_namer.dart';
+import '../transform.dart';
+import 'transform_referred_type.dart';
+
+ClassPropertyDeclaration transformProperty(
+  ClassPropertyDeclaration originalProperty,
+  ClassPropertyDeclaration wrappedClassInstance,
+  UniqueNamer globalNamer,
+  TransformationMap transformationMap,
+) {
+  final transformedType = transformReferredType(
+    originalProperty.type,
+    globalNamer,
+    transformationMap,
+  );
+
+  final transformedProperty = ClassPropertyDeclaration(
+    id: originalProperty.id,
+    name: originalProperty.name,
+    hasSetter: originalProperty.hasSetter,
+    type: transformedType,
+    hasObjCAnnotation: true,
+  );
+
+  transformedProperty.getterStatements = _generateGetterStatemenets(
+    originalProperty,
+    wrappedClassInstance,
+    transformedProperty,
+    globalNamer,
+    transformationMap,
+  );
+
+  if (originalProperty.hasSetter) {
+    transformedProperty.setterStatements = _generateSetterStatemenets(
+      originalProperty,
+      wrappedClassInstance,
+      transformedProperty,
+      globalNamer,
+      transformationMap,
+    );
+  }
+
+  return transformedProperty;
+}
+
+List<String> _generateGetterStatemenets(
+  ClassPropertyDeclaration originalProperty,
+  ClassPropertyDeclaration wrappedClassInstance,
+  ClassPropertyDeclaration transformedProperty,
+  UniqueNamer globalNamer,
+  TransformationMap transformationMap,
+) {
+  final wrappedInstanceProperty =
+      '${wrappedClassInstance.name}.${originalProperty.name}';
+
+  if (originalProperty.type.isObjCRepresentable) {
+    assert(originalProperty.type.id == transformedProperty.type.id);
+    return [wrappedInstanceProperty];
+  }
+
+  if (originalProperty.type is GenericType) {
+    throw UnimplementedError('Generic types are not implemented yet');
+  }
+
+  final transformedTypeDeclaration = transformDeclaration(
+    (originalProperty.type as DeclaredType).declaration,
+    globalNamer,
+    transformationMap,
+  );
+
+  assert(transformedTypeDeclaration.id == transformedProperty.type.id);
+
+  return ['${transformedTypeDeclaration.name}($wrappedInstanceProperty)'];
```

Comment:
You've already got logic that wraps return values in `_generateMethodStatements`. Can it be factored out into a function and shared with the logic here? The wrap function could include pretty much everything in this function after the `if (originalProperty.type.isObjCRepresentable)` check.

Same for the unwrapping logic below. That could be factorized with the arg unwrapping logic.

Reply:
Done. I actually noticed that I'm re-wrapping the `newValue` in the setter instead of unwrapping it. I fixed that too.